### DISCUSSION TITLE
How to get the vehicles in a block?

### DISCUSSION THREAD

**Discussion by sehb** [openTCS/opentcs#29] [Q&A] [answered]
How to get the vehicles in a block?

**Comment by swltr** [comment]
What situation are you facing, i.e. what exactly are you trying to achieve?

**Comment by sehb** [comment]
If the vehicle is powered off in a BLOCK, the resources occupied by the vehicle will not be automatically released by the kernel. But, the vehicle cannot be seen in the BLOCK. If we can get which vehicle in the BLOCK ,  we can solved this problem. This is our purpose. Thanks!

**Comment by swltr** [comment, marked as answer]
Powering off a vehicle somewhere does not mean it's automatically removed from the visualization. You should still be able to see it in the Operations Desk application, for instance. However, I believe what you are looking for is the integration level of a vehicle (see https://www.opentcs.org/docs/5.5/user/opentcs-users-guide.html#_vehicle and https://www.opentcs.org/docs/5.5/user/opentcs-users-guide.html#_configuring_vehicle_drivers). You can set this integration level to e.g. noticed or _ ignored_ after powering off the vehicle; the resources occupied by the vehicle are released then.

**Reply by sehb** [reply]
thanks!
